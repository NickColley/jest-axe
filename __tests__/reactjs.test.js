const React = require('react')
const ReactDOMServer = require('react-dom/server')
const { render } = require('@testing-library/react')

const { axe, toHaveNoViolations } = require('../index')

expect.extend(toHaveNoViolations)

describe('React', () => {

  test('renders correctly', async () => {
    const element = React.createElement('img', { src: '#' })
    const html = ReactDOMServer.renderToString(element)

    const results = await axe(html)
    expect(() => {
      expect(results).toHaveNoViolations()
    }).toThrowErrorMatchingSnapshot()
  })

  test('renders a react testing library container correctly', async () => {
    const element = React.createElement('img', { src: '#' })
    const { container } = render(element)
    const results = await axe(container)
    
    expect(() => {
      expect(results).toHaveNoViolations()
    }).toThrowErrorMatchingSnapshot()
  })

  test('renders a react testing library container without duplicate ids', async () => {
    const element = React.createElement('img', { src: '#', alt: 'test', id: 'test' })
    const { container } = render(element)
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
