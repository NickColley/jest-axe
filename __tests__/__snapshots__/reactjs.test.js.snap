// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React renders a react testing library container correctly 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toHaveNoViolations([22m[32mexpected[39m[2m)[22m

Expected the HTML found at $('img') to have no violations:

[90m<img src="#">[39m

Received:

[31m"Images must have alternate text (image-alt)"[39m

[33mFix any of the following:[39m
[33m  Element does not have an alt attribute[39m
[33m  aria-label attribute does not exist or is empty[39m
[33m  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty[39m
[33m  Element has no title attribute[39m
[33m  Element's default semantics were not overridden with role="none" or role="presentation"[39m

You can find more information on this issue here: 
[34mhttps://dequeuniversity.com/rules/axe/4.7/image-alt?application=axeAPI[39m"
`;

exports[`React renders correctly 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toHaveNoViolations([22m[32mexpected[39m[2m)[22m

Expected the HTML found at $('img') to have no violations:

[90m<img src="#">[39m

Received:

[31m"Images must have alternate text (image-alt)"[39m

[33mFix any of the following:[39m
[33m  Element does not have an alt attribute[39m
[33m  aria-label attribute does not exist or is empty[39m
[33m  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty[39m
[33m  Element has no title attribute[39m
[33m  Element's default semantics were not overridden with role="none" or role="presentation"[39m

You can find more information on this issue here: 
[34mhttps://dequeuniversity.com/rules/axe/4.7/image-alt?application=axeAPI[39m

â”€â”€â”€â”€â”€â”€â”€â”€

Expected the HTML found at $('img') to have no violations:

[90m<img src="#">[39m

Received:

[31m"All page content should be contained by landmarks (region)"[39m

[33mFix any of the following:[39m
[33m  Some page content is not contained by landmarks[39m

You can find more information on this issue here: 
[34mhttps://dequeuniversity.com/rules/axe/4.7/region?application=axeAPI[39m"
`;
